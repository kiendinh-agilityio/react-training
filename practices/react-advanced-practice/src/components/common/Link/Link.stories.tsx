import Link from '.';

export default {
  component: Link,
  title: 'Link',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const NavbarItem = {
  args: {
    href: '/Blog',
    children: 'Blog',
  },
};
