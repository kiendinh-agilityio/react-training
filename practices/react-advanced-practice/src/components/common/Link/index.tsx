import { Link as LinkBase, LinkProps } from '@radix-ui/themes';

type LinkBaseProps = {
  children: React.ReactNode;
  target?: '_blank' | '_self' | '_parent' | '_top';
} & LinkProps;

const Link = ({ children, href, target = '_self', className }: LinkBaseProps) => (
  <LinkBase href={href} target={target} className={`cursor-pointer ${className}`}>
    {children}
  </LinkBase>
);

export default Link;
