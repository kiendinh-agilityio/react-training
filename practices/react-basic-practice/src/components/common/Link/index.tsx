interface LinkProps {
  href: string;
  className?: string;
  target?: '_blank' | '_self' | '_parent' | '_top';
  text: string;
}

const Link = ({ href, className = '', target = '_self', text }: LinkProps) => (
  <a href={href} className={className} target={target}>
    {text}
  </a>
);

export default Link;
