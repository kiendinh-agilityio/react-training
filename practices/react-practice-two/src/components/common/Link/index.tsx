interface LinkProps {
  href: string;
  text?: string;
  target?: '_blank' | '_self' | '_parent' | '_top';
  icon?: React.ReactNode;
  className?: string;
}

const Link = ({ text, href, className = '', target = '_self', icon }: LinkProps) => (
  <a
    href={href}
    className={`font-regular text-sm leading-[54px] ${className && className}`}
    target={target}
  >
    {text && text}
    {icon && icon}
  </a>
);

export default Link;
