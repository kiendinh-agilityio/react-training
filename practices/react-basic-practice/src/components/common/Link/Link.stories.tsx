import Link from '.';

export default {
  component: Link,
  title: 'Link',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const Gmail = {
  args: {
    href: 'mailto:aurent@simmmple.com',
    text: 'laurent@simmmple.com',
  },
};
