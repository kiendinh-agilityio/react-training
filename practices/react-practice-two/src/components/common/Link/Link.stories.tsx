import Link from '.';

// import common icons
import { TwitterIcon } from '@/components/common/Icons';

export default {
  component: Link,
  title: 'Link',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const ContactLink = {
  args: {
    href: '/contact',
    text: 'Contact',
  },
};

export const SocialLink = {
  args: {
    href: 'https://twitter.com',
    icon: <TwitterIcon />,
  },
};
