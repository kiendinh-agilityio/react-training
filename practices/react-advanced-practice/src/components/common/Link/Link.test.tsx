import { render } from '@testing-library/react';

// Component Link
import Link from '.';

describe('Link Component', () => {
  it('should match with snapshot', () => {
    const { container } = render(<Link href="#" children="Blog" />);

    expect(container).toMatchSnapshot();
  });
});
