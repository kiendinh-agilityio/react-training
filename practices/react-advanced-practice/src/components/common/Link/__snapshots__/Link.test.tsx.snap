// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component should match with snapshot 1`] = `
<div>
  <a
    class="rt-Text rt-reset rt-Link rt-underline-auto cursor-pointer undefined"
    data-accent-color=""
    href="#"
    target="_self"
  >
    Blog
  </a>
</div>
`;
